File: C&C++/Leetcode/7_ReverseInteger.c
#include <main.c>

int reverse(int x)
{

    int ans = 0;

    while (x != 0)
    {

        int mod = x % 10;
        if (ans > (2147483647 / 10 - mod / 10) || ans < (-2147483648 / 10 - mod / 10))
            return 0;
        x /= 10;
        ans = ans * 10 + mod;
    }

    return ans;
}

File: Java/Leetcode/01_Array/7_ReverseInteger.java

class ReverseInt {

    public static void main(String[] args) {
        int t = reverse(Integer.MIN_VALUE - 8);
        System.out.println(t);
    }

    public static int reverse(int x) {

        int ans = 0;

        while (x != 0) {

            int mod = x % 10;
            x /= 10;
            if (ans > (Integer.MAX_VALUE / 10 - mod / 10) || ans < (Integer.MIN_VALUE / 10 - mod / 10))
                return 0;
            ans = mod + ans * 10;

        }

        return ans;

    }
}